example/Python/src/ReflexGame/ReflexGame.lf

```diff
@@ -0,0 +1,192 @@
+target Python {
+    keepalive: true,
+    files: [gui.py]
+}
+
+preamble {=
+    import multiprocessing
+    import threading
+    import random
+    def additional_time(min_time, max_time):
+        return random.randint(min_time, max_time)
+=}
+
+reactor RandomSource(min_time(2 sec), max_time(8 sec)) {
+    input another;
+    output out;
+    logical action prompt(min_time);
+    state count(0);
+    reaction(startup) {=
+        random.seed()
+    =}
+
+    reaction(prompt) -> out {=
+        self.count += 1
+        out.set(self.count)
+    =}
+
+    reaction(another) -> prompt {=
+        # schedule a prompt event 
+        prompt.schedule(additional_time(0, self.max_time - self.min_time))
+    =}
+}
+
+/*
+ * Receives key presses from the pygame process. 
+ */
+reactor GetUserInput {
+    preamble {=
+        import threading
+        def listen_for_input(self, user_response):
+            while 1:
+                try:
+                    c = self.user_input.recv()
```

Comment:
Just FYI, apparently, `recv()` blocks until something is received. There is an alternative `poll()` method that busy waits.
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.connection.Connection

So this `while 1:` might be okay here.

Reply:
I believe I made sure that all ```while 1``` loops in this program are blocking, including the pygame loop.